api/firmware/btc_test.go

```diff
@@ -37,6 +37,43 @@ import (
 
 const hardenedKeyStart = 0x80000000
 
+func p2wpkhPkScript(pubkey *btcec.PublicKey) []byte {
+	pubkeyHash := btcutil.Hash160(pubkey.SerializeCompressed())
+	addr, err := btcutil.NewAddressWitnessPubKeyHash(pubkeyHash, &chaincfg.MainNetParams)
+	if err != nil {
+		panic(err)
+	}
+	pkScript, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		panic(err)
+	}
+	return pkScript
+}
+
+func p2trPkScript(xonlyPubkey []byte) []byte {
+	addr, err := btcutil.NewAddressTaproot(xonlyPubkey, &chaincfg.MainNetParams)
+	if err != nil {
+		panic(err)
+	}
+	pkScript, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		panic(err)
+	}
+	return pkScript
+}
+
+func p2shPkScript(redeemScript []byte) []byte {
+	addr, err := btcutil.NewAddressScriptHash(redeemScript, &chaincfg.MainNetParams)
+	if err != nil {
+		panic(err)
+	}
+	pkScript, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		panic(err)
```

Comment:
why panic? can't we pass a `t testing.T` to this function and then call `t.Helper()` and finally replacing this if with a `require.NoError(t,err)`?

(same in the other two methods)